### PR TITLE
Initial mav 2 signing support [WIP - DO NOT MERGE]

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
Completely untested. Trying to open up testing of: https://github.com/PX4/Firmware/pull/7530.

You can turn on mav 2 signing and provide your key from the mavlink settings page.

Keep in mind that QGC still does not negotiate mavlink 2 on/off based on capability bits. It just turns on mav 2 if it gets incoming mav 2. So to test with this pull you'll need to turn on mav 2 from the vehicle side first.

@simonegu, @LorenzMeier  I only sort of understand the instructions for implementing mav 2 signing. So first thing would be a code review of this to see if it makes any sense.

### CODE REVIEW

**Comment by simonegu** [comment]
@DonLakeFlyer I did find something I created a PR against your branch: https://github.com/DonLakeFlyer/qgroundcontrol/pull/2 .

File: src/comm/MAVLinkProtocol.cc [hunk 1]
```diff
@@ -198,6 +210,33 @@ void MAVLinkProtocol::receiveBytes(LinkInterface* link, QByteArray b)
...
                     mavlinkStatus->flags &= ~MAVLINK_STATUS_FLAG_OUT_MAVLINK1;
+
+                    // Setup mavlink 2 signing
+                    if (_appSettings->mavlink2Signing()->rawValue().toBool()) {
+                        mavlink_setup_signing_t setupSigning;
+                        QString key = _appSettings->mavlink2SigningKey()->rawValue().toString();
+
+                        setupSigning.target_system = message.sysid;
+                        setupSigning.target_component = message.compid;
+                        setupSigning.initial_timestamp = QDateTime(QDate(2015, 1, 1)).msecsTo(QDateTime::currentDateTimeUtc()) * 100;
```
Comment (simonegu) [inline src/comm/MAVLinkProtocol.cc]: By inspecting this with the debugger I get a huge value for the starting time stamp: 8032606717800
I personally don't know if it is correct.

**Comment by DonLakeFlyer** [comment]
Sorry @simonegu. I lost this in the fray. I'll rebase and double-check timestamping.

**Comment by DonLakeFlyer** [comment]
@simonegu Rebased and merged in your PR. With repect to the time. According to this: https://github.com/ArduPilot/mavlink/blob/master/doc/MAVLink2.md the time is specified in microseconds not milliseconds. Hence the really large number. Given that, does it still look incorrect to you?

**Comment by LorenzMeier** [comment]
@simonegu Could you have a look?

**Comment by DonLakeFlyer** [comment]
This pull seems to have gone bonkers. Let me figure out what is going on.

**Pull request closed without merging**